### PR TITLE
[FLINK-25537] [JUnit5 Migration] Module: flink-core with,Package: core

### PR DESCRIPTION
**Pull request by GOODBOY008**
changes:

- Migrate module core to junit5 with core package.

### CODE REVIEW

File: flink-core/src/test/java/org/apache/flink/core/fs/local/LocalFileSystemTest.java [hunk 1]
```diff
@@ -88,66 +83,66 @@ public void testLocalFilesystem() throws Exception {
...
+        assertThat(statusforfiles).isEmpty();
 
         // check that lfs can delete directory..
         lfs.delete(pathtotmpdir, true);
 
         // double check that directory is not existent anymore..
-        assertFalse(lfs.exists(pathtotmpdir));
-        assertFalse(tempdir.exists());
+        assertThat(lfs.exists(pathtotmpdir)).isFalse();
+        assertThat(tempdir.exists()).isFalse();
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/core/fs/local/LocalFileSystemTest.java:107]:
```suggestion
        assertThat(tempdir)).doesNotExist();
```

File: flink-core/src/test/java/org/apache/flink/core/fs/local/LocalFileSystemTest.java [hunk 2]
```diff
@@ -156,37 +151,38 @@ public void testLocalFilesystem() throws Exception {
...
          */
-        assertTrue(lfs.delete(pathtotestfile1, false));
+        assertThat(lfs.delete(pathtotestfile1, false)).isTrue();
 
         // and can lfs also delete directories recursively?
-        assertTrue(lfs.delete(pathtotmpdir, true));
+        assertThat(lfs.delete(pathtotmpdir, true)).isTrue();
 
-        assertTrue(!tempdir.exists());
+        assertThat(tempdir.exists()).isFalse();
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/core/fs/local/LocalFileSystemTest.java:173]:
```suggestion
        assertThat(tempdir).doesNotExist();
```

File: flink-core/src/test/java/org/apache/flink/core/fs/local/LocalFileSystemTest.java [hunk 3]
```diff
@@ -200,61 +196,63 @@ public void testRenamePath() throws IOException {
...
 
-        final File srcFile = temporaryFolder.newFile("someFile.txt");
-        final File destFile = new File(temporaryFolder.newFolder(), "target");
+        final File srcFile = TempDirUtils.newFile(tempFolder, "someFile.txt");
+        final File destFile = new File(TempDirUtils.newFolder(tempFolder), "target");
 
         // we need to make the file non-modifiable so that the rename fails
-        Assume.assumeTrue(srcFile.getParentFile().setWritable(false, false));
-        Assume.assumeTrue(srcFile.setWritable(false, false));
+        Assumptions.assumeTrue(srcFile.getParentFile().setWritable(false, false));
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/core/fs/local/LocalFileSystemTest.java:247]: How about using assumeThat ?

File: flink-core/src/test/java/org/apache/flink/core/io/LocatableSplitAssignerTest.java [hunk 4]
```diff
@@ -20,483 +20,420 @@
...
-                    assertTrue(Arrays.equals(is.getHostnames(), noLocalHost));
-                }
+        // get all available splits
+        LocatableInputSplitAssigner ia = new LocatableInputSplitAssigner(splits);
+        LocatableInputSplit is = null;
+        for (int i = 0; i < NUM_SPLITS; i++) {
+            String host = requestingHosts[i % requestingHosts.length];
+            is = ia.getNextInputSplit(host, 0);
+            // check valid split
+            assertThat(is != null).isTrue();
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/core/io/LocatableSplitAssignerTest.java:201]:
```suggestion
            assertThat(is).isNotNull();
```

File: flink-core/src/test/java/org/apache/flink/core/memory/ByteArrayOutputStreamWithPosTest.java [hunk 5]
```diff
@@ -20,90 +20,91 @@
...
-        stream.setPosition(-1);
+    void testSetNegativePosition() {
+        Throwable exception =
+                assertThrows(
+                        IllegalArgumentException.class,
+                        () -> {
+                            stream.write(new byte[BUFFER_SIZE]);
+                            stream.setPosition(-1);
+                        });
+        assertThat(exception.getMessage()).contains("Position out of bounds");
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/core/memory/ByteArrayOutputStreamWithPosTest.java:71]: assertThatThrownBy is preferred.

File: flink-core/src/test/java/org/apache/flink/core/memory/OffHeapUnsafeMemorySegmentTest.java [hunk 6]
```diff
@@ -47,23 +47,24 @@ MemorySegment createSegment(int size, Object owner) {
...
-    @Test
-    public void testCallCleanerOnFree() {
+    @TestTemplate
+    void testCallCleanerOnFree() {
         final CompletableFuture<Void> cleanerFuture = new CompletableFuture<>();
         MemorySegmentFactory.allocateOffHeapUnsafeMemory(
                         10, null, () -> cleanerFuture.complete(null))
                 .free();
-        assertTrue(cleanerFuture.isDone());
+        assertThat(cleanerFuture.isDone()).isTrue();
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/core/memory/OffHeapUnsafeMemorySegmentTest.java:63]:
```suggestion
        assertThat(cleanerFuture).isDone();
```

File: flink-filesystems/flink-azure-fs-hadoop/src/test/java/org/apache/flink/fs/azurefs/AzureBlobRecoverableWriterTest.java [hunk 7]
```diff
@@ -46,15 +43,11 @@ public class AzureBlobRecoverableWriterTest extends AbstractRecoverableWriterTes
     private static final String ACCESS_KEY = System.getenv("ARTIFACTS_AZURE_ACCESS_KEY");
     private static final String TEST_DATA_DIR = "tests-" + UUID.randomUUID();
 
-    @BeforeClass
+    @BeforeAll
     public static void checkCredentialsAndSetup() throws IOException {
```
Comment (Jiabao-Sun) [inline flink-filesystems/flink-azure-fs-hadoop/src/test/java/org/apache/flink/fs/azurefs/AzureBlobRecoverableWriterTest.java:47]:
```suggestion
    static void checkCredentialsAndSetup() throws IOException {
```

File: flink-filesystems/flink-azure-fs-hadoop/src/test/java/org/apache/flink/fs/azurefs/AzureBlobRecoverableWriterTest.java [hunk 8]
```diff
@@ -75,7 +68,7 @@ public static void checkCredentialsAndSetup() throws IOException {
         FileSystem.initialize(conf);
     }
 
-    @AfterClass
+    @AfterAll
     public static void afterClass() {
```
Comment (Jiabao-Sun) [inline flink-filesystems/flink-azure-fs-hadoop/src/test/java/org/apache/flink/fs/azurefs/AzureBlobRecoverableWriterTest.java:72]:
```suggestion
   static void afterClass() {
```

File: flink-filesystems/flink-hadoop-fs/src/test/java/org/apache/flink/runtime/fs/hdfs/HadoopRecoverableWriterTest.java [hunk 9]
```diff
@@ -46,21 +47,19 @@ public class HadoopRecoverableWriterTest extends AbstractRecoverableWriterTest {
 
     private static Path basePath;
 
-    @BeforeClass
+    @BeforeAll
     public static void testHadoopVersion() {
```
Comment (Jiabao-Sun) [inline flink-filesystems/flink-hadoop-fs/src/test/java/org/apache/flink/runtime/fs/hdfs/HadoopRecoverableWriterTest.java:51]:
```suggestion
   static void testHadoopVersion() {
```

File: flink-filesystems/flink-hadoop-fs/src/test/java/org/apache/flink/runtime/fs/hdfs/HadoopRecoverableWriterTest.java [hunk 10]
```diff
@@ -46,21 +47,19 @@ public class HadoopRecoverableWriterTest extends AbstractRecoverableWriterTest {
...
-        Assume.assumeTrue(HadoopUtils.isMinHadoopVersion(2, 7));
+        assumeThat(HadoopUtils.isMinHadoopVersion(2, 6)).isTrue();
     }
 
-    @BeforeClass
+    @BeforeAll
     public static void verifyOS() {
```
Comment (Jiabao-Sun) [inline flink-filesystems/flink-hadoop-fs/src/test/java/org/apache/flink/runtime/fs/hdfs/HadoopRecoverableWriterTest.java:56]:
```suggestion
   static void verifyOS() {
```

File: flink-filesystems/flink-hadoop-fs/src/test/java/org/apache/flink/runtime/fs/hdfs/HadoopRecoverableWriterTest.java [hunk 11]
```diff
@@ -46,21 +47,19 @@ public class HadoopRecoverableWriterTest extends AbstractRecoverableWriterTest {
...
-        Assume.assumeTrue(
-                "HDFS cluster cannot be started on Windows without extensions.",
-                !OperatingSystem.isWindows());
+        assumeThat(OperatingSystem.isWindows()).isFalse();
     }
 
-    @BeforeClass
+    @BeforeAll
     public static void createHDFS() throws Exception {
```
Comment (Jiabao-Sun) [inline flink-filesystems/flink-hadoop-fs/src/test/java/org/apache/flink/runtime/fs/hdfs/HadoopRecoverableWriterTest.java:61]:
```suggestion
   static void createHDFS() throws Exception {
```

File: flink-filesystems/flink-hadoop-fs/src/test/java/org/apache/flink/runtime/fs/hdfs/HadoopRecoverableWriterTest.java [hunk 12]
```diff
@@ -74,7 +73,7 @@ public static void createHDFS() throws Exception {
         basePath = new Path(hdfs.getUri() + "/tests");
     }
 
-    @AfterClass
+    @AfterAll
     public static void destroyHDFS() throws Exception {
```
Comment (Jiabao-Sun) [inline flink-filesystems/flink-hadoop-fs/src/test/java/org/apache/flink/runtime/fs/hdfs/HadoopRecoverableWriterTest.java:77]:
```suggestion
   static void destroyHDFS() throws Exception {
```

File: flink-core/src/test/java/org/apache/flink/core/fs/InitOutputPathTest.java [hunk 13]
```diff
@@ -23,59 +23,52 @@
...
 
-import static org.junit.Assert.fail;
-import static org.powermock.api.mockito.PowerMockito.whenNew;
+import static org.apache.flink.util.Preconditions.checkNotNull;
+import static org.junit.Assert.assertThrows;
 
 /** A test validating that the initialization of local output paths is properly synchronized. */
-@RunWith(PowerMockRunner.class)
-@PrepareForTest(LocalFileSystem.class)
 public class InitOutputPathTest {
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/core/fs/InitOutputPathTest.java:44]:
```suggestion
class InitOutputPathTest {
```

**Review by Jiabao-Sun — COMMENTED** [review]
Thanks @GOODBOY008 for this contribution.
I left some comments.

**Comment by 1996fanrui** [comment]
> Thanks @GOODBOY008 for the quick update, LGTM now. Hi @1996fanrui, could you help double check this PR?

Thank @Jiabao-Sun for the ping, I will check it these 2 days.

File: flink-core/src/test/java/org/apache/flink/core/fs/InitOutputPathTest.java [hunk 14]
```diff
@@ -23,59 +23,52 @@
...
-            lock.set(null, new ReentrantLock(true));
-        }
+        Field modifiers = Field.class.getDeclaredField("modifiers");
+        modifiers.setAccessible(true);
+        modifiers.setInt(lock, lock.getModifiers() & ~Modifier.FINAL);
+
+        lock.set(null, new NoOpLock());
+        // in the original un-synchronized state, we can force the race to occur by using
+        // the proper latch order to control the process of the concurrent threads
+        assertThrows(FileNotFoundException.class, () -> runTest(true));
```
Comment (1996fanrui) [inline flink-core/src/test/java/org/apache/flink/core/fs/InitOutputPathTest.java:66]: using assertThatThrownBy instead.

File: flink-core/src/test/java/org/apache/flink/core/fs/SafetyNetCloseableRegistryTest.java [hunk 15]
```diff
@@ -20,26 +20,27 @@
...
 import java.io.Closeable;
+import java.io.File;
 import java.io.IOException;
 import java.util.concurrent.atomic.AtomicInteger;
 
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
 /** Tests for the {@link SafetyNetCloseableRegistry}. */
 public class SafetyNetCloseableRegistryTest
```
Comment (1996fanrui) [inline flink-core/src/test/java/org/apache/flink/core/fs/SafetyNetCloseableRegistryTest.java:37]:
```suggestion
class SafetyNetCloseableRegistryTest
```

File: flink-core/src/test/java/org/apache/flink/core/state/StateFutureTest.java [hunk 16]
```diff
@@ -35,7 +35,6 @@
 import java.util.concurrent.atomic.AtomicReference;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.assertj.core.api.Assertions.fail;
 
 /** Tests for {@link StateFuture} related implementations. */
 public class StateFutureTest {
```
Comment (1996fanrui) [inline flink-core/src/test/java/org/apache/flink/core/state/StateFutureTest.java:40]:
```suggestion
class StateFutureTest {
```

File: flink-core/src/test/java/org/apache/flink/core/fs/local/LocalFileSystemRecoverableWriterTest.java [hunk 17]
```diff
@@ -21,18 +21,18 @@
...
 import org.apache.flink.core.fs.FileSystem;
 import org.apache.flink.core.fs.Path;
+import org.apache.flink.testutils.junit.utils.TempDirUtils;
 
-import org.junit.ClassRule;
-import org.junit.rules.TemporaryFolder;
+import org.junit.jupiter.api.io.TempDir;
 
 /** Tests for the {@link LocalRecoverableWriter}. */
 public class LocalFileSystemRecoverableWriterTest extends AbstractRecoverableWriterTest {
```
Comment (1996fanrui) [inline flink-core/src/test/java/org/apache/flink/core/fs/local/LocalFileSystemRecoverableWriterTest.java:29]:
```suggestion
class LocalFileSystemRecoverableWriterTest extends AbstractRecoverableWriterTest {
```

File: flink-core/src/test/java/org/apache/flink/core/io/DefaultSplitAssignerTest.java [hunk 18]
```diff
@@ -20,105 +20,87 @@
...
+            threads[i].join(5000);
         }
+
+        // verify
+        for (int i = 0; i < NUM_THREADS; i++) {
+            assertThat(threads[i].isAlive()).isFalse();
+        }
+
+        assertThat(splitsRetrieved.get()).isEqualTo(NUM_SPLITS);
+        assertThat(sumOfIds.get()).isEqualTo(SUM_OF_IDS);
```
Comment (1996fanrui) [inline flink-core/src/test/java/org/apache/flink/core/io/DefaultSplitAssignerTest.java:101]:
```suggestion
        assertThat(splitsRetrieved).hasValue(NUM_SPLITS);
        assertThat(sumOfIds).hasValue(SUM_OF_IDS);
```

File: flink-core/src/test/java/org/apache/flink/core/io/LocatableSplitAssignerTest.java [hunk 19]
```diff
@@ -20,483 +20,420 @@
...
-        } catch (Exception e) {
-            e.printStackTrace();
-            fail(e.getMessage());
+        // verify
+        for (int i = 0; i < NUM_THREADS; i++) {
+            assertThat(threads[i].isAlive()).isFalse();
         }
+
+        assertThat(splitsRetrieved.get()).isEqualTo(NUM_SPLITS);
+        assertThat(sumOfIds.get()).isEqualTo(SUM_OF_IDS);
```
Comment (1996fanrui) [inline flink-core/src/test/java/org/apache/flink/core/io/LocatableSplitAssignerTest.java:303]:
```suggestion
        assertThat(splitsRetrieved).hasValue(NUM_SPLITS);
        assertThat(sumOfIds).hasValue(SUM_OF_IDS);
```

File: flink-core/src/test/java/org/apache/flink/core/io/LocatableSplitAssignerTest.java [hunk 20]
```diff
@@ -20,483 +20,420 @@
...
```
Comment (1996fanrui) [inline flink-core/src/test/java/org/apache/flink/core/io/LocatableSplitAssignerTest.java:362]: These 2 can be updated to hasValue as well.

File: flink-core/src/test/java/org/apache/flink/core/io/LocatableSplitAssignerTest.java [hunk 21]
```diff
@@ -20,483 +20,420 @@
...
-            }
+        // verify
+        for (int i = 0; i < NUM_THREADS; i++) {
+            assertThat(threads[i].isAlive()).isFalse();
+        }
 
-            assertEquals(NUM_SPLITS, splitsRetrieved.get());
-            assertEquals(SUM_OF_IDS, sumOfIds.get());
+        assertThat(splitsRetrieved.get()).isEqualTo(NUM_SPLITS);
+        assertThat(sumOfIds.get()).isEqualTo(SUM_OF_IDS);
```
Comment (1996fanrui) [inline flink-core/src/test/java/org/apache/flink/core/io/LocatableSplitAssignerTest.java:425]: These 2 can be updated to hasValue as well.

File: flink-core/src/test/java/org/apache/flink/core/state/StateFutureTest.java [hunk 22]
```diff
@@ -45,13 +44,13 @@ public class StateFutureTest {
             };
 
     @Test
-    public void basicSyncComplete() {
+    void basicSyncComplete() {
         StateFutureImpl.CallbackRunner runner = new TestCallbackRunner(null);
         final AtomicInteger counter = new AtomicInteger(0);
```
Comment (1996fanrui) [inline flink-core/src/test/java/org/apache/flink/core/state/StateFutureTest.java:49]: All assertThat can be updated to hasValue as well.

File: flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/endpoint/hive/HiveServer2EndpointITCase.java [hunk 23]
```diff
@@ -137,7 +137,7 @@ public class HiveServer2EndpointITCase extends TestLogger {
             new HiveServer2EndpointExtension(SQL_GATEWAY_SERVICE_EXTENSION::getService);
 
     @BeforeAll
-    public static void setup() throws Exception {
+    static void setup() throws Exception {
```
Comment (1996fanrui) [inline flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/endpoint/hive/HiveServer2EndpointITCase.java:140]: This class shouldn't extends `TestLogger`, and the public can be removed at class level.

Note: the public can be removed for all test methods.

TBH, these changes should be finished in the previous PR, but they are forgot. Thank you for the updating!

File: flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/security/token/HiveServer2DelegationTokenProviderITCase.java [hunk 24]
```diff
@@ -49,15 +49,15 @@
 public class HiveServer2DelegationTokenProviderITCase {
```
Comment (1996fanrui) [inline flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/security/token/HiveServer2DelegationTokenProviderITCase.java:49]:
```suggestion
class HiveServer2DelegationTokenProviderITCase {
```

Note: the public can be removed for all test methods.

**Review by 1996fanrui — CHANGES_REQUESTED** [review]
Thanks @GOODBOY008 for the hard work.

I left some comments, please take a look in your free time, thanks~

**Comment by 1996fanrui** [comment]
@flinkbot run azure

**Review by 1996fanrui — COMMENTED** [review]
Thanks @GOODBOY008 for the update!

LGTM except 2 comments: 

- One old comment[1] isn't addressed thoroughly, the public can be removed for all test methods of HiveServer2DelegationTokenProviderITCase class.
- It's better to rebase master branch, and let's follow the CI later.

[1] https://github.com/apache/flink/pull/24881#discussion_r1636171034

**Review by 1996fanrui — APPROVED** [review]
LGTM assuming the CI is green!

Hi @GOODBOY008 , please cc me once the CI is green, thanks

**Pull request merged**